sdk/core/Azure.Core/src/DynamicData/MutableJsonDocument.cs

```diff
@@ -76,14 +76,27 @@ private void WriteJson(Stream stream)
         {
             if (!Changes.HasChanges)
             {
-                Write(stream, _original.Span);
+                WriteOriginal(stream);
                 return;
             }
 
             using Utf8JsonWriter writer = new(stream);
             RootElement.WriteTo(writer);
         }
 
+        private void WriteOriginal(Stream stream)
+        {
+            using Utf8JsonWriter writer = new(stream);
+            if (_original.Length == 0)
+            {
+                _originalDocument.WriteTo(writer);
```

Comment:
Since we don't need writer in the `else` clause, we can move it inside `if` clause